Backend-API/src/main/java/com/avangers/backendapi/services/UserServiceImpl.java

```diff
@@ -71,4 +64,21 @@ public DeleteUserResponseDTO deleteUser(String email) {
                 .response("User was successfully deleted")
                 .build();
     }
+
+   @Override
+   public LoginResponseDTO login(LoginRequestDTO loginRequestDTO) {
+    // Retrieve the user by email
+    User user = userRepository.findByEmail(loginRequestDTO.getEmail()).orElseThrow(
+            () -> new UsernameNotFoundException("User not found")
+    );
+    
+    // Check if the provided password matches the stored password
+    if (passwordEncoder.matches(loginRequestDTO.getPassword(), user.getPassword())) {
+        // Call the token generator inside LoginResponseDTO() if needed
+        return new LoginResponseDTO();
+    }
+
+    // Throw an exception if the password is invalid
+    throw new RuntimeException("Password is not valid");
```

Comment:
The indentation of the method is off here. And (opinion alert) I am not the biggest fa of the comments here neither, but I do not mind them. I see little or no benefits, but more maintainability with them.